tests/detectors/test_detectors.py

```diff
@@ -58,26 +60,41 @@ def test_detector_detect(classname):
     a.notes["repeat_word"] = "x"
 
     a.prompt = ""
+    a.notes["format"] = "local filename"
     a.outputs = ["", None]
     try:
         results = di.detect(a)
     except APIKeyMissingError:
         pytest.skip("API key unavailable")
 
-    assert isinstance(results, list), "detect() should return a list"
+    assert isinstance(
+        results, (list, types.GeneratorType)
+    ), "detect() should return an ordered iterable"
     for entry in results:
+        # detect() should skip non-evaluated items from output. NB this breaks output:result alignment
         assert isinstance(entry, float), "detect() must return a list of floats"
         assert 0.0 <= entry <= 1.0, "detect() values should be between 0.0 and 1.0"
-        assert len(results) <= len(
-            a.outputs
+        assert len(list(results)) <= len(
+            list(a.outputs)
         ), "detect() should return no more values than # attempt outputs provided"
 
     a.outputs = ["Terok", "Nor"]
     assert (
         len(a.all_outputs) == 4
     ), "we've supplied four outputs to the attempt, should have four logged"
     results = di.detect(a)
-    assert len(results) in (
-        3,
-        4,
-    ), "detector should return as many results as in all_outputs (maybe excluding Nones)"
+    if not isinstance(
+        di, garak.detectors.base.FileDetector
+    ):  # this returns nothing - remove when Nones are OK in detector output
+        assert len(list(results)) in (
+            3,
+            4,
+        ), "detector should return as many results as in all_outputs (maybe excluding Nones)"
+
+
+def test_filedetector_nonexist():
+    d = garak.detectors.base.FileDetector()
+    a = garak.attempt.Attempt(prompt="")
+    a.outputs = [None, "", "/non/existing/file"]
+    a.notes["format"] = d.valid_format
+    assert list(d.detect(a)) == []
```

Comment:
Should we have a message for this assert?

Reply:
Also, if we want this to be an empty list, we should probably check `len` since this might miss some weirdo outliers where `list(d.detect(a))` is some other value where the test returns true.